src/private/lidar_impl.cpp

```diff
@@ -0,0 +1,191 @@
+#include "context_impl.cpp"
+#include <vector>
+#include <libm2k/utils/utils.hpp>
+#include <libm2k/utils/channel.hpp>
+#include <libm2k/utils/devicegeneric.hpp>
+#include <libm2k/utils/devicein.hpp>
+#include <libm2k/lidar.hpp>
+#include <iio.h>
+#include <iostream>
+
+using namespace std;
+using namespace libm2k::contexts;
+using namespace libm2k::utils;
+
+class Lidar::LidarImpl : public ContextImpl {
+public:
+	LidarImpl(std::string uri, iio_context* ctx, std::string name, bool sync) :
+		ContextImpl(uri, ctx, name, sync) {
+
+		pulse_dev = new DeviceGeneric(ctx, "7c700000.axi-pulse-capture");
+		afe_dev	  = new DeviceGeneric(ctx, "ad5627");
+		adc_dev	  = new DeviceIn     (ctx, "axi-ad9094-hpc");
```

Comment:
these values seem hard-coded;
for now, maybe this is fine as is, but maybe later a better dynamic mechanism would be better;
if you change the address from `7c700000` to something else, it no longer works;

Reply:
I'm not sure what you mean here. 
These are the devices available on the board. What else would one use instead of "hard-coded values"?

Reply:
We can't really determine dynamically which are the IIO devices we need, other than specifying the exact name. This looks ok to me for now. If we find some other mechanism we can do the changes later on.

It's also done here: https://github.com/analogdevicesinc/libm2k/blob/master/src/analog/private/m2kanalogin_impl.cpp#L50 where we need those 2 devices for configuration.

Reply:
> I'm not sure what you mean here.

so, the number `7c700000` comes from here https://github.com/analogdevicesinc/linux/blob/master/arch/arm/boot/dts/zynq-zc706-adv7511-fmclidar1.dts#L59

which means, that the `axi-pulse-capture` IP is mapped at that specific address;

typically, customers take our designs as reference, and change stuff; sometimes it's it in the kernel, or device-tree, or similar;
sometimes, they also change addresses;
when they stuck, they ask questions here:  https://ez.analog.com/   [and we have to answer stuff there]

i'm seeing quite a potential for some customer-questions regarding that address;

Reply:
> We can't really determine dynamically which are the IIO devices we need, other than specifying the exact name. This looks ok to me for now. If we find some other mechanism we can do the changes later on.

Fine by me: I am raising some points here about the relationship between Linux + DT & this code.
But we do need to keep in mind this, when customers ask us stuff.

Another way to do this, is via some config mechanism.
But, that is outside of the scope of this changeset.

Reply:
So, regarding this device only that address is subject to change? The device name is not subject to change, right?

If that's the case, then sure, we can do a search for the device name only, regardless of the address. Maybe it's reasonable to implement that inside the libm2k itself and not on Lidar? Meaning, you can do a `DeviceGeneric(ctx, "axi-pulse-capture")` instead and drop the address part from the object creation.

Reply:
> So, regarding this device, only that address is subject to change? The device name is not subject to change, right?

We can choose to make the device-name constant.
But the address is something that should always be treated as variable/floating,

Reply:
I just had another thought; this can be something to consider later:
```
		afe_dev	  = new DeviceGeneric(ctx, "ad5627");
		adc_dev	  = new DeviceIn     (ctx, "axi-ad9094-hpc");
```

the `ad5627` and `ad9094` names seem specific for this version of Lidar (let's say Lidar1);
but if we want to do Lidar2 [later], the parts may be different;
sometimes, some parts go obsolete [or out-of-stock], so you end up with having the replace `ad5627` with some other similar part (e.g. ad5xxx)

maybe a more generic approach would be to try to name these devices `lidar-afe` or `lidar0-afe` in the OS, so that they can be swap-able at a later point in time;

naturally, we don't need to do this now; but this is something to consider